### PR TITLE
Fix setup.py neuron-ls issue

### LINKED ISSUE #2661: Fail to use ``pip install -e .''
Here are the errors. I also installed numpy, but still failed. CUDA==12.2

```
Collecting git+https://github.com/vllm-project/vllm                                                                                                   
  Cloning https://github.com/vllm-project/vllm to /tmp/pip-req-build-yuxv0lpu                                                                         
  Running command git clone --filter=blob:none --quiet https://github.com/vllm-project/vllm /tmp/pip-req-build-yuxv0lpu                               
  Resolved https://github.com/vllm-project/vllm to commit ab406446691f289ef51d1abd8d1ff66760eda36f
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      /tmp/pip-build-env-ehqfzo6d/overlay/lib/python3.10/site-packages/torch/nn/modules/transformer.py:20: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:84.)
        device: torch.device = torch.device(torch._C._get_default_device()),  # torch.device('cpu'),
      Traceback (most recent call last):
        File "/home/ec2-user/SageMaker/conda/deekseek/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/ec2-user/SageMaker/conda/deekseek/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/ec2-user/SageMaker/conda/deekseek/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-ehqfzo6d/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-ehqfzo6d/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-ehqfzo6d/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 57, in <module>
        File "<string>", line 40, in _is_cuda
        File "<string>", line 33, in _is_neuron
        File "/home/ec2-user/SageMaker/conda/deekseek/lib/python3.10/subprocess.py", line 526, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['neuron-ls']' returned non-zero exit status 1.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

### PR DESCRIPTION
**Pull request by simon-mo**
This should fix #2661

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -382,15 +378,15 @@ def read_readme() -> str:
...
-    if _is_hip():
+    if _is_cuda():
+        with open(get_path("requirements.txt")) as f:
+            requirements = f.read().strip().split("\n")
+    elif _is_hip():
         with open(get_path("requirements-rocm.txt")) as f:
             requirements = f.read().strip().split("\n")
     elif _is_neuron():
         with open(get_path("requirements-neuron.txt")) as f:
             requirements = f.read().strip().split("\n")
```
Comment (zhuohan123) [inline setup.py:389]: Add an `else` statement here and raise an error in it?

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thanks for the work!

File: setup.py [hunk 2]
```diff
@@ -28,21 +29,16 @@
...
-    try:
-        subprocess.run(["neuron-ls"], capture_output=True, check=True)
-    except FileNotFoundError:
-        torch_neuronx_installed = False
-    return torch_neuronx_installed
-
-
-def _is_cuda() -> bool:
-    return (torch.version.cuda is not None) and not _is_neuron()
+    return shutil.which("neuron-ls") is not None
```
Comment (adirajagopal) [inline setup.py:41]: Just to let you know, I tried implementing just this line and it doesn't work on an AWS AMI as the neuron-ls package is installed, but if you're not on a neuron device then it still fails. What worked for me was to just change the errors caught on line 39 to include 'subprocess.CalledProcessError'.

Comment (simon-mo) [reply setup.py]: Thanks! updated my PR to reflect that

**Pull request merged**